sensor/tests/replay/replay_test.go

```diff
@@ -0,0 +1,300 @@
+		},
+	}
+	for name, c := range cases {
+		suite.T().Run(name, func(t *testing.T) {
+			suite.fakeCentral.ClearReceivedBuffer()
+			receivedMessagesCh := make(chan *central.MsgFromSensor, 10)
+			suite.fakeCentral.OnMessage(func(msg *central.MsgFromSensor) {
+				receivedMessagesCh <- msg
+			})
+			eventsReader := &k8s.TraceReader{
+				Source: c.k8sEventsFile,
+			}
+			err = eventsReader.Init()
+			if err != nil {
+				panic(err)
+			}
+			fm := k8s.FakeEventsManager{
+				AckChannel: ackChannel,
+				Mode:       k8s.ChannelAck,
+				Client:     suite.fakeClient,
+				Reader:     eventsReader,
+			}
+			_, errCh := fm.CreateEvents(context.Background())
+			// Wait for all the events to be processed
+			err = <-errCh
+			if err != nil {
+				panic(err)
+			}
+			// We continue to read the ackChannel to avoid blocking
+			ctx, cancelFunc := context.WithCancel(context.Background())
+			go func() {
+				for {
+					select {
+					case <-ackChannel:
+					case <-ctx.Done():
+						return
+					}
+				}
+			}()
+			// Wait for the re-sync to happen
+			time.Sleep(5 * resyncTime)
+			allEvents := suite.fakeCentral.GetAllMessages()
+			// Read the sensorOutputFile containing the expected sensor's output
+			expectedEvents, err := readSensorOutputFile(c.sensorOutputFile)
+			if err != nil {
+				panic(err)
+			}
+			expectedDeployments := getAllDeployments(expectedEvents)
+			receivedDeployments := getAllDeployments(allEvents)
+			assert.Equal(t, len(expectedDeployments), len(receivedDeployments))
+			for id, exp := range expectedDeployments {
+				if e, ok := receivedDeployments[id]; !ok {
+					t.Error("Deployment not found")
+				} else {
+					assert.Equal(t, exp.GetDeployment().GetServiceAccountPermissionLevel(), e.GetDeployment().GetServiceAccountPermissionLevel())
+					assert.Equal(t, exp.GetDeployment().GetPorts(), e.GetDeployment().GetPorts())
+					assert.Equal(t, exp.GetDeployment().GetName(), e.GetDeployment().GetName())
+					assert.Equal(t, exp.GetDeployment().GetNamespace(), e.GetDeployment().GetNamespace())
+					assert.Equal(t, exp.GetDeployment().GetLabels(), e.GetDeployment().GetLabels())
+					assert.Equal(t, exp.GetDeployment().GetPodLabels(), e.GetDeployment().GetPodLabels())
+					assert.Equal(t, exp.GetDeployment().GetImagePullSecrets(), e.GetDeployment().GetImagePullSecrets())
+				}
+			}
+			expectedPods := getAllPods(expectedEvents)
+			receivedPods := getAllPods(allEvents)
+			assert.Equal(t, len(expectedPods), len(receivedPods))
+			for id, exp := range expectedPods {
+				if e, ok := receivedPods[id]; !ok {
+					t.Error("Pod not found")
+				} else {
+					assert.Equal(t, exp.GetPod().GetDeploymentId(), e.GetPod().GetDeploymentId())
+					assert.Equal(t, exp.GetPod().GetName(), e.GetPod().GetName())
+					assert.Equal(t, exp.GetPod().GetNamespace(), e.GetPod().GetNamespace())
+				}
+			}
+			cancelFunc()
+		})
+	}
+}
+
+func readSensorOutputFile(fname string) ([]*central.MsgFromSensor, error) {
```

Comment:
Pure dark magic here :) 

I guess this code is vulnerable to changes to the proto definition of the `MsgFromSensor`. How can we prevent it going out of sync?

Reply:
It is vulnerable to changes. I don't know how to work around that. I think the entire suit of tests is sensitive to changes in the proto definitions. If we change something there, we can expect the tests might stop working.

Reply:
I think it makes sense, I also don't remember why this wouldn't be a good idea. Maybe we can open a follow-up task and investigate. What do you think?

Reply:
Sounds good

Reply:
Created task for this [ROX-11566](https://issues.redhat.com/browse/ROX-11566)